lib/Controller/SettingController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace OCA\FaceRecognition\Controller;
+
+use OCA\FaceRecognition\Db\FaceMapper;
+use OCA\FaceRecognition\Db\ImageMapper;
+use OCA\FaceRecognition\Db\PersonMapper;
+
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\IRequest;
+use OCP\IConfig;
+use OCP\IUserManager;
+use OCP\IUser;
+
+class SettingController extends Controller {
+
+	/** @var IConfig */
+	private $config;
+
+	/** @var IUserManager */
+	private $userManager;
+
+	/** @var FaceMapper */
+	private $faceMapper;
+
+	/** @var ImageMapper */
+	private $imageMapper;
+
+	/** @var PersonMapper */
+	private $personMapper;
+
+	/** @var string */
+	private $userId;
+
+	public function __construct ($appName,
+	                             IRequest     $request,
+	                             IConfig      $config,
+	                             IUserManager $userManager,
+	                             FaceMapper   $faceMapper,
+	                             ImageMapper  $imageMapper,
+	                             PersonMapper $personMapper,
+	                             $userId)
+	{
+		parent::__construct($appName, $request);
+		$this->appName      = $appName;
+		$this->config       = $config;
+		$this->userManager  = $userManager;
+		$this->faceMapper   = $faceMapper;
+		$this->imageMapper  = $imageMapper;
+		$this->personMapper = $personMapper;
+		$this->userId       = $userId;
+	}
+
+	/**
+	 * @param $type
+	 * @param $value
+	 * @return JSONResponse
+	 * @throws \OCP\PreConditionNotMetException
+	 */
+	public function setAppValue($type, $value) {
+		$this->config->setAppValue('facerecognition', $type, $value);
+		switch ($type) {
+			case 'sensitivity':
+				$this->userManager->callForSeenUsers(function(IUser $user) {
+					$this->config->setUserValue($user->getUID(), 'facerecognition', 'recreate-clusters', 'true');
+				});
+				break;
+			default:
+				break;
+		}
+
+		$result = [
+			'success' => 'true',
```

Comment:
probably bikesheding at this point, but why is 'success' true if $value was not 'sensitivity' (if it fall through case switch).

P.S. we need something like S_FALSE in C :)

Reply:
> probably bikesheding at this point, but why is 'success' true if $value was not 'sensitivity' (if it fall through case switch).

I suppose that as we add preferences we can improve this.

> P.S. we need something like S_FALSE in C :)

Maybe I can pass the Boolean directly. I do not remember why I used string here.

Reply:
Ahh..
> Why is 'success' true if $value was not 'sensitivity'

The setting always changes, you can send garbage that changes it, and that's why it returns ok .. the switch comes to consider the special cases, when you have to do something else when changing a particular option. In this case, recreate the clusters